### PR TITLE
added shogun::memcpy wrapper for std::memcpy

### PR DESCRIPTION
**Pull request by lambday**
Needs rebase.

### CODE REVIEW

File: tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #include <thread>
 
 #ifdef HAVE_VIENNACL
-#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#include <shogun/mathematics/linalg/LinalgBackendViennacl.h>
```
Comment (lambday) [inline tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc]: @OXPHOS :P did you change the name of the file recently?

File: tests/unit/lib/memory_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+{
+	shogun::memcpy(dest, src, size);
+}
+
+TEST(memory, memcpy)
+{
+	index_t m=10;
+	index_t d=5;
+	SGMatrix<float64_t> src(d, m);
+	for (index_t i=0; i<d*m; ++i)
```
Comment (vigsterkr) [inline tests/unit/lib/memory_unittest.cc]: i wish we would could put this into a utility part of the unit tests... would it be too much to ask to do it? we keep repeating these type of codes all around unit tests.

File: tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc (hunk 1 above)
Comment (vigsterkr) [inline tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc]: @lambday she merged linalg refactor yesterday into develop :) i think this slipped everybody :)

Comment (OXPHOS) [inline tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc]: @lambday Sorry I believe it's a bug..It doesn't distinguish cases when I run the codes at local so I didn't notice..

Comment (OXPHOS) [inline tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc]: @vigsterkr Can I just fix it quickly by pushing it directly to develop branch? Or you want to wait till when I look into other problems in ViennaCL? Or @lambday just fix it here?

Comment (vigsterkr) [inline tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc]: @OXPHOS push the fix directly to develop and @lambday will rebase over that. :)

File: tests/unit/lib/memory_unittest.cc (hunk 2 above)
Comment (lambday) [inline tests/unit/lib/memory_unittest.cc]: should be a separate PR.. maybe an entrance task?

File: tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc (hunk 1 above)
Comment (lambday) [inline tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc]: umm.. why not merge this PR instead as it fixes the issue? It will have to wait till another 8 hours before I could rebase and merge this PR.. although there is no hurry..

Comment (vigsterkr) [inline tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc]: @lambday no hurry... prefer the camelized notation over that weird mixture of camelized and not...

Comment (lambday) [inline tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc]: haha true that :D

File: tests/unit/lib/memory_unittest.cc (hunk 2 above)
Comment (karlnapf) [inline tests/unit/lib/memory_unittest.cc]: maybe put the utility method and then make an entrance task for people to port other tests to using that

File: tests/unit/lib/Memory_unittest.cc [hunk 3]
```diff
@@ -54,3 +55,27 @@ TEST(MemoryTest,SGMatrix)
...
+{
+	shogun::memcpy(dest, src, size);
+}
+
+TEST(MemoryTest, memcpy)
+{
+	const index_t size = 10;
+	auto src = SG_CALLOC(float64_t, size);
+	for (index_t i=0; i<size; ++i)
+		src[i]=CMath::randn_double();
```
Comment (lambday) [inline tests/unit/lib/Memory_unittest.cc]: Will do the util thing in a separate PR.

File: tests/unit/lib/memory_unittest.cc (hunk 2 above)
Comment (lambday) [inline tests/unit/lib/memory_unittest.cc]: @karlnapf cool.. will do that in a separate PR...

**Comment by lambday** [comment]
travis errors seem unrelated. merging this one.
